### PR TITLE
[Issue 13854][broker] Fix call sync method in async rest api for internalSetReplicatedSubscriptionStatus

### PR DESCRIPTION
**Pull request by suiyuzeng**
Master Issue: #13854

### Motivation
See #13854

### Verifying this change

- [x] Make sure that the change passes the CI checks.

### Does this pull request potentially affect one of the following parts:

*If `yes` was chosen, please highlight the changes*

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API: (no)
  - The schema: (no)
  - The default values of configurations: (no)
  - The wire protocol: (no)
  - The rest endpoints: (no)
  - The admin cli options: (no)
  - Anything that affects deployment: (no)

### Documentation
  
- [x] `no-need-doc`

### CODE REVIEW

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java [hunk 1]
```diff
@@ -4577,12 +4575,10 @@ protected void internalSetReplicatedSubscriptionStatus(AsyncResponse asyncRespon
...
-            String subName, boolean authoritative, boolean enabled) {
-        try {
-            // Redirect the request to the appropriate broker if this broker is not the owner of the topic
-            validateTopicOwnership(topicName, authoritative);
-
+    private CompletableFuture<Void> internalSetReplicatedSubscriptionStatusForNonPartitionedTopicAsync(AsyncResponse asyncResponse,
+        String subName,boolean authoritative,boolean enabled) {
+        // Redirect the request to the appropriate broker if this broker is not the owner of the topic
+        return validateTopicOwnershipAsync(topicName, authoritative).thenRun(() -> {
             Topic topic = getTopicReference(topicName);
```
Comment (mattisonchao) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java:4582]: This method can be async.

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java [hunk 2]
```diff
@@ -4504,26 +4504,24 @@ protected void internalSetReplicatedSubscriptionStatus(AsyncResponse asyncRespon
...
-            validateTopicOperation(topicName, TopicOperation.SET_REPLICATED_SUBSCRIPTION_STATUS, subName);
-        } catch (Exception e) {
-            resumeAsyncResponseExceptionally(asyncResponse, e);
-            return;
-        }
+        // 1.Permission to consume this topic is required
+        // 2.Redirect the request to the peer-cluster if the local cluster is not included in the replication clusters
+        CompletableFuture<Void> future =
+                validateTopicOperationAsync(topicName, TopicOperation.SET_REPLICATED_SUBSCRIPTION_STATUS, subName)
+                        .thenCompose(__-> validateGlobalNamespaceOwnershipAsync(namespaceName));
```
Comment (mattisonchao) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java:4511]:
```suggestion
                        .thenCompose(__ -> validateGlobalNamespaceOwnershipAsync(namespaceName));
```

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java [hunk 3]
```diff
@@ -4577,12 +4575,10 @@ protected void internalSetReplicatedSubscriptionStatus(AsyncResponse asyncRespon
...
     }
 
-    private void internalSetReplicatedSubscriptionStatusForNonPartitionedTopic(AsyncResponse asyncResponse,
-            String subName, boolean authoritative, boolean enabled) {
-        try {
-            // Redirect the request to the appropriate broker if this broker is not the owner of the topic
-            validateTopicOwnership(topicName, authoritative);
-
+    private CompletableFuture<Void> internalSetReplicatedSubscriptionStatusForNonPartitionedTopicAsync(AsyncResponse asyncResponse,
+        String subName,boolean authoritative,boolean enabled) {
```
Comment (mattisonchao) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java:4579]:
```suggestion
        String subName, boolean authoritative, boolean enabled) {
```

Comment (codelipenghui) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java:4579]:
```suggestion
    private CompletableFuture<Void> internalSetReplicatedSubscriptionStatusForNonPartitionedTopicAsync(AsyncResponse asyncResponse,
        String subName, boolean authoritative, boolean enabled) {
```

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java [hunk 4]
```diff
@@ -4603,16 +4599,18 @@ private void internalSetReplicatedSubscriptionStatusForNonPartitionedTopic(Async
...
+                        topicName,
                         subName);
                 asyncResponse.resume(Response.noContent().build());
             } else {
                 asyncResponse.resume(new RestException(Status.METHOD_NOT_ALLOWED,
                         "Cannot enable/disable replicated subscriptions on non-persistent topics"));
             }
-        } catch (Exception e) {
+        }).exceptionally(e -> {
             resumeAsyncResponseExceptionally(asyncResponse, e);
```
Comment (mattisonchao) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java:4611]: - Maybe we need ``e.getCause`` here.
- We need log some information when get exception.

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java (hunk 1 above)
Comment (codelipenghui) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java:4582]: Nice catch!

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java [hunk 5]
```diff
@@ -4565,8 +4563,8 @@ protected void internalSetReplicatedSubscriptionStatus(AsyncResponse asyncRespon
                         return null;
                     });
                 } else {
-                    internalSetReplicatedSubscriptionStatusForNonPartitionedTopic(asyncResponse, subName, authoritative,
-                            enabled);
+                    internalSetReplicatedSubscriptionStatusForNonPartitionedTopicAsync(asyncResponse, subName,
```
Comment (Jason918) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java:4566]: No need to add "Async", let's keep it consistent with other methods.

**Review by Jason918 — APPROVED** [review]

**Comment by Jason918** [comment]
@liudezhi2098 @mattisonchao PTAL

**Review by mattisonchao — APPROVED** [review]

**Review by codelipenghui — APPROVED** [review]

**Pull request merged**